<?xml version="1.0"?>

<launch>
  <!-- start up world -->
  <param name="/use_sim_time" value="true" />
  <node name="gazebo" pkg="gazebo" type="gazebo" args="-r $(find gazebo_worlds)/worlds/empty.world" respawn="false" output="screen"/>
  <param name="wg_walls"      textfile="$(find gazebo_worlds)/objects/wg_walls.model" />
  <node name="xml2factor_wg_walls" pkg="gazebo" type="spawn_model" args="-gazebo -param wg_walls -x -25.65 -y -25.65 -model wg_walls" respawn="false" output="screen" />

  <!-- start up robot, tuck arms for navigation -->
  <include file="$(find pr2_gazebo)/pr2.launch"/>
  <node pkg="pr2_tuckarm" type="tuck_arms.py" args="b" output="screen" name="tuck_arms"/>

  <!-- start up navigation stack -->
  <include file="$(find pr2_machine)/sim.machine" />
  <include file="$(find pr2_navigation_teleop)/teleop.xml" />
  <include file="$(find pr2_navigation_perception)/lasers_and_filters.xml" />
  <include file="$(find pr2_navigation_perception)/ground_plane.xml" />
  <include file="$(find pr2_navigation_global)/move_base.xml" />

  <!-- dynamic map generation -->
  <node name="gmapping_node" pkg="gmapping" type="slam_gmapping" respawn="false" >
    <remap to="base_scan" from="scan"/>
    <param name="odom_frame" value="odom_combined" />
  </node>

  <!-- start up rviz -->
  <!--<include file="$(find pr2_navigation_global)/rviz/rviz_move_base.launch"/>-->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find hide_n_seek)/hide_n_seek.vcg">
  </node>
  <node pkg="costmap_2d" type="costmap_2d_markers" name="voxel_visualizer">
    <remap from="voxel_grid" to="move_base_node/local_costmap/voxel_grid_throttled"/>
  </node>

</launch>
